nitrates/analysis_seeds/UpperLimits.py

```diff
@@ -35,21 +63,604 @@ def cli():
+    
+            for n in range(len(head)):
+
+    
+            
+                spec = head[n][0]
+                time_bin = head[n][1]
+                
+                logging.info('ul weighting %s %s' %(spec, time_bin))
+  
+
+                if time_bin=='0.128':
+
+                    ul_5sigma_new = ul_arr['%s' %spec]
+
+
+                    def f_ul_n(x,y):
+                        return f_ul(x,y,ras,decs,ul_5sigma_new)
+        
+                    dp=0
+                    norm=0
+        
+                    for i in range(npix):
+        
+                        theta, phi = hp.pix2ang(nside, i)  # Retrieve theta and phi angles
+                        dec = np.degrees(np.pi / 2 - theta)  # Convert theta to Dec
+                        ra = np.degrees(phi)   # Convert phi to RA
+        
+        
+                        if map_values[i]>0:
+        
+                            ul=f_ul_n(ra,dec)
+                            dp+=ul*map_values[i] 
+                            norm+=map_values[i]
+
+
+                    df.at[time_bin,spec] = "{:.2e}".format(dp/norm)
+
+                    ul_final['%s' %spec]=dp/norm
+                
+                else:
+
+                    df.at[time_bin,spec] = "{:.2e}".format(ul_final['%s' %spec]*ratio['%s' %spec]['%s' %time_bin])
+
+    
+                if spec=='normal' and time_bin=='1.024':
+
+                    ul_5sigma_new= np.array(ul_arr['%s' %spec]) * ratio['%s' %spec]['%s' %time_bin]
+
+                    if fermi:
+                        plot_ul_map(ras,decs,ul_5sigma_new, skymap_file , ra_ea, red_ea, radius_ear, tri_name, path_results)
+                    else:
+                        plot_ul_map(ras,decs,ul_5sigma_new,'None', ra_ea, red_ea, radius_ear, tri_name, path_results)
+
+    
+                
+    
+            #logging.info('check post ul')
+
+            df.to_csv(os.path.join(path_results,'ul.txt'), sep='\t', index=True)
+
+            logging.info('ul done')
+
+
 
 def main(args):
+
+    work_dir=args.work_dir
+
+    file_path = os.path.join(work_dir,'config.json')
+
+    with open(file_path, 'r') as json_file:
+        json_data = json.load(json_file)
+    
+    trig_time=json_data['trigtime']
+
+    trig_name = '%s_c%s' %(json_data['triggerID'], json_data['config']['id'])
+    path_results='/gpfs/group/jak51/default/nitrates_all_results/UL/%s' %trig_name
+
+    if os.path.exists(path_results)== False:
+        'done'
+        os.mkdir(path_results)
+
     logging.basicConfig(
-        filename="upper_limits_analysis.log",
+        filename=os.path.join(path_results,'upper_limits_analysis.log'),
         level=logging.DEBUG,
         format="%(asctime)s-" "%(levelname)s- %(message)s",
     )
 
-    work_dir=args.work_dir
+
     #resp_dname=args.resp_dname
 
     conn = get_conn(os.path.join(work_dir,'results.db'))
```

Comment:
do you need his?

Reply:
gihub is being supid. im referring to line 666, get_conn

Reply:
that is the part of Gayathri, I think it's needed to get the trigger time

Reply:
its not actually called again...

Reply:
conn is called here
https://github.com/Swift-BAT/NITRATES/blob/a48a8b74c511ea3e4a04dc7422f4db0942932dfb/nitrates/analysis_seeds/UpperLimits.py#L664C1-L668C35